### ISSUE TITLE
Number of Stars vs Error

### ISSUE THREAD

**Issue by CentofantiEze** [tobias-liaudat/wf-SEDs#12]
### Generating the dataset
- [x] Generate datasets with `n_stars = [10k, 9k,...,2k, 1k]` using `n_bins = 8` and `pupil_diameter = 128`.
- [x] Train the model with each dataset, with `n_bins = 8` and `pupil_diameter = 128`.
- [x] Generate a plot displaying the number of stars used in each training on the x-axis and the relative SR PSF error on the y-axis.

### Analyse the resulting models
- [x] Plot the learnt DD features and compare with other model's DD features (`n_bins = 20` and `pupil_diameter = 256`).

### Training
- [x] Define the values of the learning rate and the number of epochs for each dataset. We want to maintain a kind of constant number of gradient updates (or passes) so when we increase the number of stars in the dataset  we adjust the number of epochs in the training. See Table C1 from the wavefront paper to have an idea.

**Comment by tobias-liaudat** [comment]
# New experience: test with more complex models using 4k stars

We use the same setting as before, but now we focus on studying how the complexity of the model plays an impact on the performance in a scenario with high star density.

- [x] Define the number of epochs to use with the current model in the case of 4k stars in the training dataset. Eventually, change the learning rate if it seems not to be optimal.


### List of complex models to test

Current setting: 
- `model =poly`, `d_max_nonparam=5`, `n_zernikes =15` and `d_max=2` (21 DD features)

New settings
- [x] sanity check: `model =poly`, `d_max_nonparam=5`, `n_zernikes=15` and `d_max=2`
- [x] `model =poly`, `d_max_nonparam=7`, `n_zernikes=15` and `d_max=2`  (36 DD features)
- [x] `model =poly`, `d_max_nonparam=10`, `n_zernikes=15` and `d_max=2` (66 DD features)
- [x] `model = mccd `, `d_max_nonparam=5`, `graph_features=15`, `n_zernikes=15` and `d_max=2` (36 DD features)
- [x] `model = mccd `, `d_max_nonparam=5`, `graph_features=29`, `n_zernikes=15` and `d_max=2` (50 DD features)
- [x] `model = mccd `, `d_max_nonparam=7`, `graph_features=14`, `n_zernikes=15` and `d_max=2` (50 DD features)
- [x] `model = mccd `, `d_max_nonparam=7`, `graph_features=30`, `n_zernikes=15` and `d_max=2` (66 DD features)

### New test changing the optimisation strategy

Let us now focus on one of the previous models:
- `model =poly`, `d_max_nonparam=10`, `n_zernikes=15` and `d_max=2` (66 DD features)

Solving issue #17  solves the experience points below.

- [x] To do this we first need to solve the [issue 25 ](https://github.com/tobias-liaudat/wf-psf/issues/25) from the wf-psf repo. 

We would like to evaluate its performance under two different training conditions:
- [x] Test many cycles (e.g. 10) with few epochs in each (e.g. the total number of epochs we use now (cycle1 + cycle2) divided by the total number of cycles (10). We would like to have the same total number of epochs as before. The epoch numbers should be adjusted so that in the last cycle there are twice or three times as many epochs as in the previous ones. Regarding the learning rate of each cycle, I would set for the first 4 cycles the learning rate we used in the original first cycle. Then, decrease the learning rate (linearly perhaps) so that the last of the new cycles has the same learning rate as the original second cycle that we did before.
- [x] Learn the model only using the non-parametric part using the same number of cycles and epochs.

### Documentation
- [x] present results as a table in the Internship Notes

**Comment by tobias-liaudat** [comment]
Done :)
